gems/sorbet-runtime/test/types/props/serializable.rb

```diff
@@ -494,26 +494,275 @@ class CustomTypeStruct < T::Struct
     prop :hash_both, T::Hash[CustomType, CustomType], default: {}
   end
 
+  class CustomTypeWrapper
+    include T::Props::Serializable
+    prop :struct, CustomTypeStruct
+  end
+
   describe 'custom type' do
     it 'round trips as plain value' do
       assert_equal('foo', CustomTypeStruct.from_hash({'single' => 'foo'}).serialize['single'])
     end
 
+    it 'raises serialize errors for members with a custom subtype' do
+      struct = CustomTypeStruct.new
+      struct.instance_variable_set(:@single, 'not a serializable thing')
+      e = assert_raises(TypeError) do
+        struct.serialize
+      end
+
+      assert_includes(e.message, "Expected type T::Props::CustomType")
+    end
+
+    # It's hard to write tests for a CustomType with a custom deserialize method, so skip that.
+
+    it 'raises serialize errors for members with a serializable subtype' do
+      struct = CustomTypeWrapper.new
+      struct.instance_variable_set(:@struct, 'not a serializable thing')
+      e = assert_raises(NoMethodError) do
+        struct.serialize
+      end
+
+      assert_includes(e.message, "undefined method `serialize'")
+    end
+
+    it 'raises deserialize errors for members with a serializable subtype' do
+      obj = 'not a serializable thing'
+      e = assert_raises(TypeError) do
+        CustomTypeWrapper.from_hash({'struct' => obj})
+      end
+
+      assert_includes(e.message, 'provided to from_hash')
+    end
+
     it 'round trips as array value' do
       assert_equal(['foo'], CustomTypeStruct.from_hash({'array' => ['foo']}).serialize['array'])
     end
 
+    it 'raises serialize errors for members needing map' do
+      obj = CustomType.new
+      struct = CustomTypeStruct.new
+      struct.instance_variable_set(:@array, obj)
+      e = assert_raises(NoMethodError) do
+        struct.serialize
+      end
+
+      assert_includes(e.message, "undefined method `map'")
+    end
+
+    it 'raises deserialize errors for members needing map' do
+      msg_string = nil
+      extra_hash = nil
+      T::Configuration.soft_assert_handler = proc do |msg, extra|
+        msg_string = msg
+        extra_hash = extra
+      end
+
+      obj = CustomType.new
+      result = CustomTypeStruct.from_hash({'array' => obj})
+      assert_equal(obj, result.array)
+
+      refute_nil(msg_string)
```

Comment:
Can we say anything more about `msg_string` here? (I'm assuming it would say something about the fact that `map` is missing, though I may not understand what it is.)

Reply:
We can't say anything super-useful about `msg_string` (it's always "Deserialization error (probably unexpected stored type)").  I suppose we could assert something about `storytime[:error]` a little further on.  I'll do that instead.